tests/test_remove_mon_from_cluster.py

```diff
@@ -0,0 +1,91 @@
+"""
+A Testcase to remove mon from
+when I/O's are happening.
+
+Polarion-ID- OCS-355
+
+"""
+
+import logging
+import re
+import random
+import time
+from ocs import ocp, constants
+from ocsci.testlib import tier4, ManageTest
+from ocsci.config import ENV_DATA
+from resources import pod
+from tests.helpers import ceph_health_check, run_io_with_rados_bench
+
+
+log = logging.getLogger(__name__)
+
+DEP = ocp.OCP(
+    kind=constants.DEPLOYMENT, namespace=ENV_DATA['cluster_namespace']
+)
+POD = ocp.OCP(
+    kind=constants.POD, namespace=ENV_DATA['cluster_namespace']
+)
+
+
+def get_mons_from_cluster():
+    """
+    Getting the list of mons from the cluster
+
+    Returns:
+        available_mon (list): Returns the mons from the cluster
+    """
+
+    ret = DEP.get(
+        resource_name='', out_yaml_format=False, selector='app=rook-ceph-mon'
+    )
+    available_mon = re.findall(r'[\w-]+mon-+[\w-]', ret)
+    return available_mon
+
+
+def remove_mon_from_cluster(mon):
+    """
+    Removing the mon pod from deployment
+    """
+    assert DEP.delete(resource_name=mon), f"Failed to delete mon {mon}"
+
+
+def verify_mon_pod_up():
+    """
+    Verify mon pods are in Running state.
+
+    Returns:
+    """
+    log.info(f"Verifying all mons pods are up and Running")
+    time.sleep(15)
+    ret = POD.wait_for_resource(
+        condition=constants.STATUS_RUNNING, selector='app=rook-ceph-mon',
+        resource_count=3, timeout=700)
+    log.info(f"waited for all mon pod to come up and running {ret}")
+    return ret
+
+
+def run_io_on_pool():
+    tools_pod = pod.get_ceph_tools_pod()
+    tools_pod.add_role(role='client')
+
+    return run_io_with_rados_bench(
+        ceph_pods=[tools_pod],
+        config={'time': 20, 'cleanup': False}
+    )
+
+
+@tier4
+class TestOcs355(ManageTest):
+
+    def test_remove_mon_pod_from_cluster(self):
+        """
+        To remove mon pod from the cluster
+
+        """
+        list_mons = get_mons_from_cluster()
+        assert run_io_on_pool(), f'Failed to run I/Os on the pool'
+        ceph_health_check()
+        if len(list_mons) > 1:
+            remove_mon_from_cluster(random.choice(list_mons))
```

Comment:
At the end of testcase (or in fixture) there should be teardown that adds mon back to cluster so the state of cluster is the same as it was at the beginning of test.

Reply:
@fbalak After removing the mon, the operator will keep on searching for the removed mon (say mon-a) on the node. It waits and if it does not find mon-a the operator will create a new mon (say mon-d ) on its own. I don't think we will explicitly have to add mon, the operator will do it in this case.

Reply:
Ah, ok. Good point.